src/app/api/[[...route]]/routes/sharehouses.ruote.ts

```
@@ -1,14 +1,64 @@
 import { Hono } from 'hono';
 
-export const app = new Hono();
+import prisma from '@/lib/prisma';
+import { auth } from '@/lib/auth';
+import type { IAssignedData } from '@/utils/assigned-data';
+
+const app = new Hono();
 
 export default app;
 
-app.get('/', (c) => {
-  return c.json({ message: 'Share houses route' });
-});
+app.get('/', async (c) => {
+  const session = await auth();
+
+  if (!session) {
+    return c.json({
+      error: 'Unauthorized',
+    });
```

Could you return an error code as well?

---

I added the 401 error code.